### ISSUE TITLE
Inconsistency whether percentile iterators should return time

### ISSUE THREAD

**Issue by andyfeller** [influxdata/influxdb#7634]
### Bug report

__System info:__ InfluxDB 1.1

__Additional info:__ 

Unsure whether this is actually a bug or not, however I noticed the inconsistency when I was trying to see how percentile() handles null or missing data.  I thought it better to ask whether an issue than ignore it.  Forgive for being unable to develop a process to confirm if an actual bug or not.

1. [float percentile iterator returning time](https://github.com/influxdata/influxdb/blob/master/influxql/call_iterator.go#L1108)
```
/ NewFloatPercentileReduceSliceFunc returns the percentile value within a window.
func NewFloatPercentileReduceSliceFunc(percentile float64) FloatReduceSliceFunc {
	return func(a []FloatPoint) []FloatPoint {
		length := len(a)
		i := int(math.Floor(float64(length)*percentile/100.0+0.5)) - 1

		if i < 0 || i >= length {
			return nil
		}

		sort.Sort(floatPointsByValue(a))
		return []FloatPoint{{Time: a[i].Time, Value: a[i].Value, Aux: a[i].Aux}}
	}
}
```
2. [integer percentile iterator returning ZeroTime](https://github.com/influxdata/influxdb/blob/master/influxql/call_iterator.go#L1123)
```
// NewIntegerPercentileReduceSliceFunc returns the percentile value within a window.
func NewIntegerPercentileReduceSliceFunc(percentile float64) IntegerReduceSliceFunc {
	return func(a []IntegerPoint) []IntegerPoint {
		length := len(a)
		i := int(math.Floor(float64(length)*percentile/100.0+0.5)) - 1

		if i < 0 || i >= length {
			return nil
		}

		sort.Sort(integerPointsByValue(a))
		return []IntegerPoint{{Time: ZeroTime, Value: a[i].Value, Aux: a[i].Aux}}
	}
}
```

**Comment by andyfeller** [comment]
Seems like this relates to #6292 [commit hash 66a599825b7afaa031604b1ef5bfca5ef7a4c0e5](https://github.com/influxdata/influxdb/commit/66a599825b7afaa031604b1ef5bfca5ef7a4c0e5) as only floats were update. @jsternberg thoughts?

**Comment by jsternberg** [comment]
That definitely looks like a bug. Percentile should be a selector for all types.

@jwilder should this fix target 1.1 or 1.2?
